libkpod/image/copy.go

```diff
@@ -152,9 +158,20 @@ func PullImage(store storage.Store, imgName string, allTags, quiet bool, sc *typ
 				}
 			}
 		}
-	} else if splitArr[0] == "oci" {
-		// needs to be implemented in future
-		return errors.Errorf("oci not supported")
+	} else if splitArr[0] == "oci-archive" {
+		// generate image reference with specified image:tag
+		if len(options.OciImage) > 0 {
+			srcRef, err = ociarchive.NewReference(splitArr[len(splitArr)-1], options.OciImage)
+			if err != nil {
+				return errors.Wrapf(err, "error parsing image name %q", options.OciImage)
+			}
+		}
+		// retrieve the manifest from index.json to access the image name
+		manifest, err := ociarchive.LoadManifestDescriptor(srcRef)
+		if err != nil {
+			return errors.Wrapf(err, "error loading manifest for %q", srcRef)
+		}
+		images = append(images, manifest.Annotations["org.opencontainers.image.ref.name"])
```

Comment:
This will set `images` to `""` if `Annotations` is missing, or does not have that annotation. Is that expected?

Reply:
For  now, the Annotations cannot be missing because oci-archive doesn't allow the user to save an image with an empty ref.name.

Reply:
What about archives generated by other tools?

Reply:
Hmm, I didn't think about that. So if Annotations is not set, should it save the image as "<none>"?

Reply:
How would the user access to the image later? Would we print an image ID to stdout, and make the output somehow machine-parseable for use in scripts? It could probably be made to work, but it’s not quite trivial.

I think just refusing such archives with an error would be good enough, for this version. We can always add support for previously rejected input later, based on actual uses encountered in the wild.

For now I've been mostly worried about the existing code going down some unexpected path which could be difficult to recover from.

But actually looking at `storageTransport.ParseStoreReference`, it immediately rejects `""`, so the current implementation fails cleanly enough.  Detecting the situation at this point would still be useful to give the user an informative error message (“the archive does not have a name annotation, don’t know what name to use to store the image”) instead of the current `"invalid reference"`.

Reply:
okay, that makes sense. I will fix that, thanks :)

Reply:
fixed.